service/tracker/execution.go

```diff
@@ -171,6 +189,48 @@ func (e *Execution) processNext() error {
 	return nil
 }
 
+func (e *Execution) checkTrieUpdates(record *uploader.BlockData, blockID flow.Identifier) error {
+	e.log.Debug().Hex("block_id", blockID[:]).Msg("fetching updates from data sync")
+	e.log.Debug().Int("updates", len(record.TrieUpdates)).Msg("got trie updates from GCP")
+	// get Trie updates from exec data sync
+	req := &access.GetExecutionDataByBlockIDRequest{BlockId: blockID[:]}
+	res, err := e.execClient.GetExecutionDataByBlockID(context.Background(), req)
+	if err != nil {
+		return fmt.Errorf("could not get execution data from access node: %w", err)
+	}
+
+	execTrieUpdates := make(map[string]bool, 0)
+	// collect updates to map to compare
+	for _, gcpUpdate := range record.TrieUpdates {
+		if gcpUpdate != nil && !gcpUpdate.IsEmpty() {
+			key := gcpUpdate.String()
+			execTrieUpdates[key] = true
+			e.log.Debug().Hex("blockID", blockID[:]).Str("update", key).Msg("got update from GCP")
+		}
+	}
+	// hash search for matching updates in exec data API
+	for _, chunk := range res.GetBlockExecutionData().ChunkExecutionData {
```

Comment:
This could return nil, better to separate: 
```go
exeData := res.GetBlockExecutionData()
if exeData == nil {}
for _, chunk := range exeData. ChunkExecutionData { ... }
```